common/persistence/nosql/nosqlplugin/cassandra/tasks.go

```diff
@@ -0,0 +1,492 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+// Portions of the Software are attributed to Copyright (c) 2020 Temporal Technologies Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	p "github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin"
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra/gocql"
+	"github.com/uber/cadence/common/types"
+)
+
+const (
+	// Row types for table tasks
+	rowTypeTask = iota
+	rowTypeTaskList
+)
+
+const (
+	taskListTaskID = -12345
+	initialRangeID = 1 // Id of the first range of a new task list
+)
+
+const (
+	templateTaskListType = `{` +
+		`domain_id: ?, ` +
+		`name: ?, ` +
+		`type: ?, ` +
+		`ack_level: ?, ` +
+		`kind: ?, ` +
+		`last_updated: ? ` +
+		`}`
+
+	templateTaskType = `{` +
+		`domain_id: ?, ` +
+		`workflow_id: ?, ` +
+		`run_id: ?, ` +
+		`schedule_id: ?,` +
+		`created_time: ? ` +
+		`}`
+
+	templateCreateTaskQuery = `INSERT INTO tasks (` +
+		`domain_id, task_list_name, task_list_type, type, task_id, task) ` +
+		`VALUES(?, ?, ?, ?, ?, ` + templateTaskType + `)`
+
+	templateCreateTaskWithTTLQuery = `INSERT INTO tasks (` +
+		`domain_id, task_list_name, task_list_type, type, task_id, task) ` +
+		`VALUES(?, ?, ?, ?, ?, ` + templateTaskType + `) USING TTL ?`
+
+	templateGetTasksQuery = `SELECT task_id, task ` +
+		`FROM tasks ` +
+		`WHERE domain_id = ? ` +
+		`and task_list_name = ? ` +
+		`and task_list_type = ? ` +
+		`and type = ? ` +
+		`and task_id >= ? ` +
+		`and task_id < ?`
```

Comment:
Is this intentional? It used to be (Exclusive, Inclusive].

Reply:
Yes, that’s intentional for consistency

Reply:
Yeah, maybe just let it be inconsistent. I just found that the queueCRUD also has an API using exclusive,inclusive manner. 

I will change it back.